src/features/tweaks/restore_attribution_links.js

```diff
@@ -25,31 +29,57 @@ const listenerOptions = { capture: true };
 const processPosts = async function (postElements) {
   postElements.forEach(async postElement => {
     const {
+      authorBlog,
       blogName,
+      community,
       id,
       postUrl,
       rebloggedFromId,
       rebloggedFromName,
-      rebloggedFromUrl
+      rebloggedFromUrl,
     } = await timelineObject(postElement);
+    const postAttributionName = (!!community && authorBlog?.name) || blogName;
+
     const postAttributionLink = postElement.querySelector(postAttributionLinkSelector);
     const reblogAttributionLink = postElement.querySelector(reblogAttributionLinkSelector);
+    const trailItemElements = [...postElement.querySelectorAll(trailItemSelector)];
 
-    if (postAttributionLink && postAttributionLink.textContent === blogName) {
+    if (postAttributionLink && postAttributionLink.textContent === postAttributionName) {
       postAttributionLink.dataset.originalHref ??= postAttributionLink.getAttribute('href');
-      postAttributionLink.href = postUrl;
       postAttributionLink.dataset.blogName = blogName;
       postAttributionLink.dataset.postId = id;
+      postAttributionLink.href = postUrl;
       postAttributionLink.addEventListener('click', onLinkClick, listenerOptions);
     }
 
     if (reblogAttributionLink && reblogAttributionLink.textContent === rebloggedFromName) {
       reblogAttributionLink.dataset.originalHref ??= reblogAttributionLink.getAttribute('href');
-      reblogAttributionLink.href = rebloggedFromUrl;
       reblogAttributionLink.dataset.blogName = rebloggedFromName;
       reblogAttributionLink.dataset.postId = rebloggedFromId;
+      reblogAttributionLink.href = rebloggedFromUrl;
       reblogAttributionLink.addEventListener('click', onLinkClick, listenerOptions);
     }
+
+    trailItemElements.forEach(async trailItemElement => {
+      const { blog, post, isContributedContent } = await trailItem(trailItemElement);
+      const trailAttributionLink = trailItemElement.querySelector(trailAttributionLinkSelector);
+
+      if (trailAttributionLink && trailAttributionLink.textContent === blog?.name && post?.id) {
+        trailAttributionLink.dataset.originalHref ??= trailAttributionLink.getAttribute('href');
+        trailAttributionLink.dataset.blogName = blog.name;
+        trailAttributionLink.dataset.postId = post.id;
+
+        if (isContributedContent) {
+          trailAttributionLink.href = postUrl;
+        } else if (blog.url.startsWith('https://www.tumblr.com/')) {
+          trailAttributionLink.href = `${blog.url.replace(/\/$/, '')}/${post.id}`;
+        } else {
+          trailAttributionLink.href = `${blog.url.replace(/\/$/, '')}/post/${post.id}`;
+        }
```

Comment:
```suggestion
          trailAttributionLink.href = postUrl;
        } else {
          const pathname = new URL(blog.url).origin === 'https://www.tumblr.com' ? `/${post.id}` : `/post/${post.id}`;
          trailAttributionLink.href = new URL(pathname, blog.url);
        }
```

...use native URL parsing, right? (But yeah, I would also be tempted to just let Tumblr handle it; we already are doing that with the `tumblr.com/@[blogname]` trick.)